### DISCUSSION TITLE
Serving Static Files without a file extension, prevent download?

### DISCUSSION THREAD

**Discussion by maitham** [vercel/next.js#10648] [Help] [answered]
I'm trying to serve my apple-app-site-association using the next server. I place the file in the public folder but rather than it being served, it gets downloaded. Is there a way to serve it without using a custom server?

I'm assuming this is down to the fact that there is no file extension so the next-server doesn't know what content type to add to the header, so it defaults to document. Is there a way to change the headers, or another way to serve this.

**Comment by Timer** [comment, marked as answer]
Next.js uses the extension to automatically detect the `content-type` of the file. Since the file is extensionless, it's served as binary content—this **should** be compatible with Apple's verification.

If Apple requires specific headers, you can overwrite this type:
```js
// next.config.js
module.exports = {
  experimental: {
    headers() {
      return [
        {
          source: "/.well-known/apple-app-site-association",
          headers: [{ key: "content-type", value: "application/json" }]
        }
      ];
    }
  }
};
```

**Reply by maitham** [reply]
Do I need to do anything special to enable experimental features? I've updated next this still causes a download of the file.

**Reply by ijjk** [reply]
Hi, you need to be on the latest canary of Next.js for this feature to work properly `v9.2.3-canary.17`. To verify the header is being set start next with `yarn next` and check for the header in the output of `curl -v http://localhost:3000/.well-known/apple-app-site-association`

**Reply by maitham** [reply]
Yea apologies was testing in dev. Works once it's built. Thanks!

**Reply by mhtamun** [reply]
This works for me in the local, but not in production @Timer

**Reply by ekimcem** [reply]
I think when you use export build the headers are not applied in next config

**Comment by barclayd** [comment]
Using next.js 9.5, it can be achieved in 2 steps:

1. Put your `apple-app-site-assocation` within `/public/.well-known/apple-app-site-assocation`
2. Add the following next.config.js:
```
module.exports = {
  async headers() {
    return [
      {
        source: '/.well-known/apple-app-site-association',
        headers: [
          {
            key: 'Content-Type',
            value: 'application/json',
          },
        ],
      },
    ];
  },
};
```

The browser will now render the apple-app-site-association as json without downloading it. Can also be verified by doing this command within the terminal:
`curl -v http://localhost:3000/.well-known/apple-app-site-association` - looking for the response body returned and `Content-Type` header in the respnse

**Reply by rdewolff** [reply]
This works perfectly well. Thanks for sharing the solution @barclayd ! 🥳

**Reply by mhtamun** [reply]
This works for me in the local, but not in production

**Reply by AlejandroGutierrezB** [reply]
Same here, any solution?

**Reply by pjindal91** [reply]
Same issue. Does not work in production

**Reply by tgwow** [reply]
Same here, local is OK, but in prod just work if i put a bar (/) in the end of filename. This probably is causing issue when validating AASA. This is my folder structure:
![folder structure](https://user-images.githubusercontent.com/28604537/137379456-244cccc4-4e6c-4733-a7b7-f0fea0ae867b.png)

**Reply by saranya20-t** [reply]
I'm also facing the same issue in production. In local, it's working fine

**Reply by tgwow** [reply]
If you are hosting on nginx server, this config works for me:
```nginx
location /.well-known/apple-app-site-association {
    alias /usr/share/nginx/html/.well-known;
    index apple-app-site-association;
    autoindex on;
    default_type application/json;
}

**Reply by aliosmanerkan** [reply]
@tgwow solution works well for me and @mhtamun fix this by serving his next.js app using an express server in the production.

**Reply by peter-stuart-turner** [reply]
also doesn't work on production for us - this is quite urgent for us

**Reply by bas080** [reply]
> 

Are you using `next export` on production? In that case you need to use the tool that serves the static files to set headers.

**Reply by peter-stuart-turner** [reply]
Have gotten it working using this in vercel.json:
{
  "headers": [
    {
      "source": "/.well-known/apple-app-site-association",
      "headers": [
        {
          "key": "Content-Type",
          "value": "application/json"
        }
      ]
    }
  ]
}

**Reply by meetmakwana7396** [reply]
Thanks! This worked for me.

**Comment by ghost** [comment]
Is there any plan to support `.well-known/*` as a special metadata files within the app router?
